Source/WebCore/loader/FrameLoader.cpp

```diff
@@ -1219,6 +1219,32 @@ static NavigationNavigationType determineNavigationType(FrameLoadType loadType,
     return NavigationNavigationType::Push;
 }
 
+// https://html.spec.whatwg.org/multipage/browsing-the-web.html#url-and-history-update-steps
+void FrameLoader::updateURLAndHistory(const URL& newURL, RefPtr<SerializedScriptValue>&& stateObject, NavigationHistoryBehavior historyHandling)
+{
+    ASSERT(m_frame->document());
+
+    if (documentLoader()->isInitialAboutBlank())
```

Comment:
What guarantees `documentLoader()` is non-null?

Reply:
The assumption I think is that the previous document exists and before committing to the new one we perform this algorithm so the document loader should be available. I added a check to the ASSERT since I think the (previous) document is in a strange state if there is no document loader available.